### DISCUSSION TITLE
Deprecated without any notice

### DISCUSSION THREAD

**Discussion by cxlblm** [laravel/framework#39856] [General] [answered]
I don't understand much about this [pr](https://github.com/laravel/framework/pull/39219), we don't deprecate conversions to exceptions at this point, for the user I never know what method, function I'm using is deprecated until one day they become real errors. We should expose deprecation notifications in the testing, local environment to find out the existence of the problem at development time without affecting the production environment, giving us more time to deal with it.

**Comment by nunomaduro** [comment, marked as answer]
If you would like to log these deprecation warnings, you may specify your preferred deprecations log channel in your application's `config/logging.php`configuration file: [laravel.com/docs/8.x/logging#logging-deprecation-warnings](https://laravel.com/docs/8.x/logging#logging-deprecation-warnings).

**Reply by cxlblm** [reply]
Thanks for your reply, I have seen this in the documentation. But I personally feel that the default approach of the framework is just to make the problem not felt, not that the problem does not exist, and if someone is not very concerned about PHP deprecation, it has become more serious by the time the problem really arises. Personally, I think we should actively solve the problems that arise, rather than blocking him out until he becomes a bigger problem.
